### ISSUE TITLE
Missing dependency in ASR

### ISSUE THREAD

**Issue by Pranavchiku** [lfortran/lfortran#3504]
```fortran
pure function my_merge(val1, val2, mask) result(merged_value)
  integer, intent(in) :: val1, val2
  logical, intent(in) :: mask
  integer :: merged_value

  if (mask) then
    merged_value = val1
  else
    merged_value = val2
  end if

end function my_merge


program main
contains
function mean_2_cdp_cdp(x, dim, mask) result(res)
  complex, intent(in) :: x(:,:)
  integer, intent(in) :: dim
  logical, intent(in), optional :: mask
  complex :: res(my_merge(size(x, 1), size(x, 2), mask=1<dim))

  interface
    pure function my_merge(val1, val2, mask) result(merged_value2)
      integer, intent(in) :: val1, val2
      logical, intent(in) :: mask
      integer :: merged_value2
    end function my_merge
  end interface

end function mean_2_cdp_cdp
end program

```

```console
% lfortran a.f90 --show-asr
ASR verify pass error: Function mean_2_cdp_cdp depends on my_merge but isn't found in its dependency list.
  --> a.f90:17:1 - 31:28
   |
17 |    function mean_2_cdp_cdp(x, dim, mask) result(res)
   |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
...
   |
31 |    end function mean_2_cdp_cdp
   | ...^^^^^^^^^^^^^^^^^^^^^^^^^^^^ failed here


Note: Please report unclear or confusing messages as bugs at
https://github.com/lfortran/lfortran/issues.
```

**Comment by Pranavchiku** [comment]
Potential fix is through PR #3388.

**Comment by Pranavchiku** [comment]
Can you please check this with #3504, it shall now work :)
